packages/uploadthing/src/next.ts

```diff
@@ -1,73 +1,13 @@
 import type { Json } from "@uploadthing/shared";
-import { getStatusCodeFromError, UploadThingError } from "@uploadthing/shared";
 
-import { UPLOADTHING_VERSION } from "./constants";
-import { defaultErrorFormatter } from "./internal/error-formatter";
-import type { RouterWithConfig } from "./internal/handler";
-import {
-  buildPermissionsInfoHandler,
-  buildRequestHandler,
-} from "./internal/handler";
-import type { FileRouter, inferErrorShape } from "./internal/types";
 import type { CreateBuilderOptions } from "./internal/upload-builder";
 import { createBuilder } from "./internal/upload-builder";
+import { createServerHandler } from "./server";
 
 export type { FileRouter } from "./internal/types";
 
 export const createUploadthing = <TErrorShape extends Json>(
   opts?: CreateBuilderOptions<TErrorShape>,
 ) => createBuilder<"app", TErrorShape>(opts);
 
-export const createNextRouteHandler = <TRouter extends FileRouter>(
-  opts: RouterWithConfig<TRouter>,
-) => {
-  const requestHandler = buildRequestHandler<TRouter, "app">(opts);
-
-  const POST = async (req: Request) => {
-    const response = await requestHandler({ req });
-    const errorFormatter =
-      opts.router[Object.keys(opts.router)[0]]?._def.errorFormatter ??
-      defaultErrorFormatter;
-
-    if (response instanceof UploadThingError) {
-      const formattedError = errorFormatter(
-        response,
-      ) as inferErrorShape<TRouter>;
-      return new Response(JSON.stringify(formattedError), {
-        status: getStatusCodeFromError(response),
-        headers: {
-          "x-uploadthing-version": UPLOADTHING_VERSION,
-        },
-      });
-    }
-    if (response.status !== 200) {
-      // We messed up - this should never happen
-      return new Response("An unknown error occured", {
-        status: 500,
-        headers: {
-          "x-uploadthing-version": UPLOADTHING_VERSION,
-        },
-      });
-    }
-
-    return new Response(JSON.stringify(response.body), {
-      status: response.status,
-      headers: {
-        "x-uploadthing-version": UPLOADTHING_VERSION,
-      },
-    });
-  };
-
-  const getBuildPerms = buildPermissionsInfoHandler<TRouter>(opts);
-
-  const GET = () => {
-    return new Response(JSON.stringify(getBuildPerms()), {
-      status: 200,
-      headers: {
-        "x-uploadthing-version": UPLOADTHING_VERSION,
-      },
-    });
-  };
-
-  return { GET, POST };
-};
+export const createNextRouteHandler = createServerHandler;
```

Comment:
this doesn't actually need to exist, we could alias it to `createServerHandler` and potentially deprecate it in favor of

```ts
import { createServerHandler } from "uploadthing/server";
import { createUploadThing } from "uploadthing/next";
```

or do you think we should keep it? no reason to special treat next here really...

Reply:
actually we did need this to be it's own thing since Next only accepts a very strict signature on the exports and the request being typed as `Request | { request: Request }` wasn't very welcome.